src/types.ts

```diff
@@ -231,8 +234,8 @@ export interface IEmailProperties {
   bodyValues: {
     [bodyPartId: string]: IEmailBodyValue;
   } | null;
-  textBody: IEmailBodyPart[] | null;
-  htmlBody: IEmailBodyPart[] | null;
+  textBody: Partial<IEmailBodyPart>[] | null;
```

Comment:
The thing is that textBody and htmlBody have properties that dont have to be set on creating the email.

I am not that good with typescript, so this is the only solution i came up with.

Maybe there is a better way. Dynamically selecting if using Partial<IEmailBodyPart>

Reply:
I don't think there is such mechanism to exclude optional or null values with TypeScript. I am mostly certain, that it needs to be JavaScript logic, that can provide this. Something, that will scan passed in object and throw everything it is allowed to throw out.
I don't think TypeScript will be even able to provide information of what is optional and what is not for such mechanism.
It mostly handles all of it syntax internally for type checking.

Reply:
I think I forgot to set most properties as optional in IEmaillBodyPart (according to JMAP specs), and if I understood correctly, you added the partial because of this.

Reply:
Yes, thats what i did